File: exs/mundo_1/ruby/032.rb
=begin
Desafio 032

Problema: Faça um programa que leia um ano qualquer e mostre se ele é bissexto.

Resolução do problema:
=end

print"Digite um ano qualquer: "
ano = gets.chomp.to_i

if ano == 0
	ano = Time.now.year
end

if (ano % 4 == 0 and ano % 100 != 0 or ano % 400 == 0)
	puts"O ano #{ano} é bissexto."
else
	puts"O ano #{ano} não é bissexto."
end

File: exs/mundo_1/golang/032.go
/*
Desafio 032

Problema: Faça um programa que leia um ano qualquer e mostre se ele é bissexto.

Resolução do problema:
*/

package main

import(
	"fmt"
	"time"
)

var ano int

func main(){
	fmt.Prtin("Digite um ano: ")
	fmt.Scanln(&ano)

	if ano == 0{
		ano = time.Now().Year()
	}

	if ano % 4 == 0 && ano % 100 != 0 || ano % 400 == 0{
		fmt.Printf("O ano %v é bissexto.\n", ano)
	} else {
		fmt.Printf("O ano %v não é bissexto.\n", ano)
	}
}
